13_sandboxes/test_case_generator.py

```diff
@@ -0,0 +1,274 @@
+# ---
+# cmd: ["modal", "run", "-m", "13_sandboxes.test_case_generator::main"]
+# ---
+import modal
+
+app = modal.App(name="sandbox-test-case-generator")
+model_volume = modal.Volume.from_name("deepseek-model-volume", create_if_missing=True)
+files_volume = modal.Volume.from_name("files-volume", create_if_missing=True)
+
+
+model_image = (
+    modal.Image.from_registry("nvidia/cuda:12.4.0-devel-ubuntu22.04", add_python="3.12")
+    .pip_install(
+        "transformers==4.53.2",
+        "torch==2.7.1",
+        "accelerate==1.8.1",
+        "hf_transfer==0.1.9",
+    )
+    .env(
+        {
+            "HF_HUB_ENABLE_HF_TRANSFER": "1",
+            "HF_HOME": "/cache",
+        }
+    )
+    .entrypoint([])  # silence noisy logs
+)
+
+with model_image.imports() as imports:
+    from transformers import AutoModelForCausalLM, AutoTokenizer
+
+
+# GH Repo Configs
+GH_OWNER = "modal-labs"
+GH_REPO_NAME = "password-analyzer"
+GH_MODULE_NAME = "password_strength"
+GH_BRANCH = "main"
+
+
+@app.cls(
+    image=model_image,
+    volumes={
+        "/cache": model_volume,
+        "/data": files_volume,
+    },
+    gpu="L40S",
+)
+class Deepseek:
+    @modal.enter()
+    def load_model(self):
+        MODEL_NAME = "deepseek-ai/deepseek-coder-6.7b-instruct"
+        REVISION = "e5d64addd26a6a1db0f9b863abf6ee3141936807"
+        self.tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME, revision=REVISION)
+        self.model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, revision=REVISION)
+        self.model.to("cuda")
+
+    def load_inputs(self, file_name: str) -> tuple[str, str]:
+        import os
+
+        if not os.path.exists("/data/inputs"):
+            raise Exception(
+                "Inputs directory does not exist. Make sure to run download_files_to_volume first."
+            )
+
+        with open(f"/data/inputs/{file_name}", "r") as f:
+            file_contents = f.read()
+
+        with open(f"/data/inputs/test_{file_name}", "r") as f:
+            test_file_contents = f.read()
+        return file_contents, test_file_contents
+
+    def write_outputs(self, output_file_name: str, output_contents: str) -> str:
+        import os
+
+        os.makedirs("/data/outputs", exist_ok=True)
+        with open(f"/data/outputs/{output_file_name}", "w") as f:
+            f.write(output_contents)
+        print(f"Output written to {output_file_name}")
+        return output_file_name
+
+    @modal.method()
+    def generate(self, file_name: str) -> str:
+        file_contents, test_file_contents = self.load_inputs(file_name)
+
+        system_prompt = get_system_prompt()
```

Comment:
Are these two components really fixed? I would expect some parts to be variable and so better to pass in via a command-line interface or similar.